### ISSUE TITLE
CollisionShape2D Polygon Shape doesn't respect parent X/Y position coordinates 

### ISSUE THREAD

**Issue by wallynm** [godotengine/godot#86942]
### Tested versions

4.2.1 stable

### System information

Godot v4.2.1.stable.mono - macOS 14.1.1 - Vulkan (Mobile) - integrated Apple M1 - Apple M1 (8 Threads)

### Issue description

Godot CollisionPolygon2D doesn't respect X/Y positions applied at Node while calculating positions using _Draw method:
```
public override void _Draw()
{
  // Godot drawn polyline with option "Editable children" for access to localPolygon updated values
  Vector2[] localPolygon = _collisionShape.Polygon.ToArray();
  localPolygon = localPolygon.Append(localPolygon[0]).ToArray();
  DrawPolyline(localPolygon, _pathColor, 40f, true);
}
```

If the Polygon is created without moving X/Y axis, `_collisionShape.Polygon.ToArray();` gets all X/Y value of points correctly, but if i move the Node positions X/Y, the Polygon Node doesn't listen to these values and as result draw the lines at incorrect positions.

Theres a full project with the functional issue running.

### Steps to reproduce

1. Create a `Area2D`
2. As children of the `Area2D` add a `CollisionPolygon2D`
3. Attach a Polygon to the `CollisionPolygon2D`
4. Implement a Draw method using Polygon Node from the recent created `CollisonPolygon2D`
5. Move the parent `CollisionPolygon2D`axis X/Y
6. The Polygon objects doesn't get updated with the X/Y values from it's parent Node.

### Minimal reproduction project (MRP)

https://github.com/wallynm/godot-issue-collision2d

**Comment by kleonc** [comment]
This is correct behavior, it works as intended.
- [`CollisionPolygon2D.polygon`](https://docs.godotengine.org/en/4.2/classes/class_collisionpolygon2d.html#class-collisionpolygon2d-property-polygon) is local to the given CollisionPolygon2D. I guess this can be clarified in the docs.
- [`CanvasItem.draw_X` methods by default draw locally to the given CanvasItem](https://docs.godotengine.org/en/4.2/tutorials/2d/custom_drawing_in_2d.html#coordinates).

You would need to somehow apply a proper transform from one coordinate space to the other. E.g. by transforming all the points or (simpler) by applying an additional transform for drawing such CollisionPolygon2D-local points.

Since these are in parent-child relation you could simply use `DrawSetTransformMatrix(_collisionShape.transform)` before the `DrawPolyline(...)` call in the Area2D (this C# version might need tweaking).

In more general case: for two CanvasItems `A` and `B` being within the same canvas layer, to get a transform converting from `A` local coordinate space to `B` local coordinate space you could use `B.get_global_transform().affine_inverse() * A.get_global_transform()`. See also [the docs](https://docs.godotengine.org/en/4.2/tutorials/2d/2d_transforms.html#transform-functions).
(for transforming a single `Vector2` you could use `B.`[`to_local`](https://docs.godotengine.org/en/4.2/classes/class_node2d.html#class-node2d-method-to-local)`(A.`[`to_global`](https://docs.godotengine.org/en/4.2/classes/class_node2d.html#class-node2d-method-to-global)`(some_A_local_point))`)

**Comment by AThousandShips** [comment]
Fixed in: #87024
